### DISCUSSION TITLE
Single-use function parameter enums

### DISCUSSION THREAD

**Discussion by alexfevery** [dotnet/csharplang#4792] [Language Ideas]
For when you need an enum for a function parameter but will only use it one time in your program.  

```
 static void Main()
 {
         Greet("John", Spanish);
 }

static void Greet(string name, enum language{English,Spanish,Japanese})
{
       if(language == English){Console.WriteLine("Hello "+name+"!");}
       if(language == Spanish){Console.WriteLine("Hola "+name+"!");}
       if(language == Japanese){Console.WriteLine("こんにちは　"+name+"さん!");}
}
```

**Comment by theunrepentantgeek** [comment]
_Every_ **enum** begins as something only needed in one place. 

But I'm unconvinced that _any_ **enum** stays that way for long. 

For example, `Greet()` would promptly be followed by `Farewell()`.

**Comment by HaloFour** [comment]
> but will only use it one time in your program.

You're using it more than once in your program, at the site where it's declared and at the site where it is called.

**Comment by CyrusNajmabadi** [comment]
Why not have : GreetSpanish, GreetEnglish, etc.?

I don't get the point of the enum that the caller had to use to customize behavior where the impl just switches on it. Why not just avoid the indirection?

**Reply by yaakov-h** [reply]
cake?

**Reply by CyrusNajmabadi** [reply]
Caller :-)

**Reply by HaloFour** [reply]
The cake was a lie.  😞
